deployment/terraform/modules/ec2/main.tf

```diff
@@ -347,3 +347,21 @@ resource "aws_elb" "https" {
     unhealthy_threshold = 4
   }
 }
+
+resource "aws_cloudwatch_metric_alarm" "cpu_utilization" {
+  alarm_name                = "cpu-utilization-limit-exceeded"
+  comparison_operator       = "GreaterThanOrEqualToThreshold"
+  evaluation_periods        = "5"
+  metric_name               = "CPUUtilization"
+  namespace                 = "AWS/EC2"
+  period                    = "300"
```

Comment:
If I'm not mistaken this is going to raise an alarm if CPU utilization has been >=25% for over 25 minutes. Is that what we want?

Reply:
This is a judgement call as far as I can tell.  25 minutes seemed like a good amount of time to allow for spikes/prolonged periods of usage.

Reply:
I see. Also, how and to whom this alarm notification will be delivered?

Reply:
I was under the impression that a notification would go to the root account user, but that was clearly mistaken.  Not sure how I came to that conclusion.  In any case: https://www.terraform.io/docs/providers/aws/r/sns_topic_subscription.html#email

In short, setting up an email endpoint for an sns topic entirely within Terraform is impossible.  My vote would be to create the topic/subscription outside Terraform within AWS and just pass in the arn as a default option to the alarms for subscription.

Reply:
Yes, email subscriptions require an out-of-band confirmation. Also, some deployments may need a different type of notification or no notification at all. Could you set it up such that it is possible to set up a notification but it is not required by default?

Reply:
Yes I think that makes sense